messaging/src/main/java/org/axonframework/messaging/MessageTypeResolver.java

```diff
@@ -36,5 +38,20 @@ public interface MessageTypeResolver {
      * @param payload The {@link Message#getPayload() Message payload} to resolve a {@link MessageType type} for.
      * @return The {@link MessageType type} for the given {@code payload}.
      */
-    <P> MessageType resolve(P payload);
+    default MessageType resolve(Object payload) {
```

Comment:
Note that this is almost exactly what the `ClassBasedMessageTypeResolver` does! :o